hicommonwealth/commonwealth
# Fix gate check logic

## Link to Issue
Closes: #5420

## Description of Changes
- Applies logic that permits gated action if member fulfills requirements for at least one group
- Applies logic that makes a member part of a group if the minimum number of requirements are met– if the "required requirements" metadata is set
- Added groupId to refreshMembership payload to distinguish results

## "How We Fixed It"


## Test Plan
- Confirm that actions (create thread, create comment, create reaction) cannot be performed unless
  - User address is a member of at least 1 group associated with the topic
- Confirm that the `metadata.required_requirements` option works correctly on a group

## Deployment Plan
N/A

## Other Considerations
N/A
